### PR TITLE
[ROCM] add custom op support, test=develop

### PR DESCRIPTION
**Pull request by qili93**
### PR types

New features

### PR changes

Others

### Describe

Revert PR https://github.com/PaddlePaddle/Paddle/pull/34050 add custom op support for ROCM.

Verification steps:

1. Exactly same code with CUDA custom op such as `setup_cuda.py`

https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/07_new_op/new_custom_op_cn.html

```python
from paddle.utils.cpp_extension import CUDAExtension, setup

setup(
    name='custom_setup_ops',
    ext_modules=CUDAExtension(
        sources=['relu_cuda.cc', 'relu_cuda.cu']
    )
)
```

2. `python setup_cuda.py install`

image

3. run custom op test with test.py content

image

Also unit test result on ROCM here:

image

### CODE REVIEW

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 1]
```diff
@@ -401,15 +401,18 @@ def unix_custom_single_compiler(obj, src, ext, cc_args, extra_postargs,
             # with dict type is dangerous.
             src = os.path.abspath(src)
             cflags = copy.deepcopy(extra_postargs)
+            print("0------------ cflags=", cflags)
```
Comment (chenwhql) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: 这个print提示是确定要保留的吗

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 2]
```diff
@@ -401,15 +401,18 @@ def unix_custom_single_compiler(obj, src, ext, cc_args, extra_postargs,
...
             try:
                 original_compiler = self.compiler.compiler_so
-                # nvcc compile CUDA source
+                print("original_compiler=", original_compiler)
```
Comment (chenwhql) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: 同上

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 1 above)
Comment (qili93) [reply python/paddle/utils/cpp_extension/cpp_extension.py]: 忘记删 debug code了，已删，谢谢！

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 2 above)
Comment (qili93) [reply python/paddle/utils/cpp_extension/cpp_extension.py]: 忘记删 debug code了，已删，谢谢！

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**